test/multiverse/suites/curb/curb_test.rb

```diff
@@ -145,6 +145,116 @@ def test_method_with_tracing_passes_the_verb_downstream
     assert Curl::Easy.new.method(:to_s).call.is_a?(String), 'Failed to create #to_s method'
   end
 
+  def test_a_noticeable_error_is_created_on_failure_if_a_segment_exists
+    multi = Curl::Multi.new
+    request = OpenStruct.new
+    error = [OpenStruct.new(name: 'name'), 'last']
+    prep_on_failure(request)
+    def multi.remove_failure_callback(request); end
+    segment = MiniTest::Mock.new
+    segment.expect :notice_error, nil, [NewRelic::Agent::NoticeableError]
+    multi.install_failure_callback(request, nil, segment)
+    segment.verify
+  end
+
+  # this doesn't assert anything directly, but confirms that something undesired
+  # did not take place, and exercises a branch of the method being tested
+  def test_a_noticeable_error_is_not_created_on_failure_if_a_segment_is_absent
+    multi = Curl::Multi.new
+    request = OpenStruct.new
+    error = [OpenStruct.new(name: 'name'), 'last']
+    prep_on_failure(request)
+    def multi.remove_failure_callback(request); end
+    # if noticeable error creation attempt, explode
+    NewRelic::Agent::NoticeableError.stub(:new, -> { raise 'kaboom' }) do
+      multi.install_failure_callback(request, nil, nil)
+    end
+  end
+
+  # we need to verify that the original callback does not receive #call if it
+  # is nil. this test does so and exercises a branch of the method being tested.
+  # should the nil check fail, this test will fail, even though it does not
+  # assert or refute anything explicitly
+  def test_an_original_callback_is_not_invoked_when_absent
+    multi = Curl::Multi.new
+    request = OpenStruct.new
+    error = [OpenStruct.new(name: 'name'), 'last']
+    prep_on_failure_without_original_callback(request)
+    def multi.remove_failure_callback(request); end
+    multi.install_failure_callback(request, nil, nil)
+  end
+
+  # if the guard against installing already-installed callbacks were to fail,
+  # the 'request' mock would receive a second _nr_instrumented call and error
+  # out. so while there is no explicit assert/refute being performed, the
+  # desired behavior is being verified
+  def test_callbacks_are_not_installed_again_if_they_have_already_been_installed
+    multi = Curl::Multi.new
+    request = MiniTest::Mock.new
+    segment = MiniTest::Mock.new
+    segment.expect :add_request_headers, nil, [MiniTest::Mock]
+    request.expect :_nr_instrumented, true
+    def multi.wrap_request(req)
+      wrapped_request = MiniTest::Mock.new
+      wrapped_request.expect :type, nil
+      wrapped_request.expect :uri, nil
+      wrapped_request.expect :method, nil
+      [wrapped_request, nil]
+    end
+    NewRelic::Agent::Tracer.stub :start_external_request_segment, segment do
+      multi.hook_pending_request(request)
+    end
+    request.verify
```

Comment:
I'm still trying to wrap my head around the `Minitest::Mock#verify` method. Why would we call `#verify` on `request`, but not on `segment`?

Reply:
The omission of a `#verify` call on `segment` here was just an oversight on my part. Added it with 953b230bece1a2d450474b8e0d77ce9802010dcb.

MiniTest (unlike RSpec and perhaps unlike Mocha) will not automatically verify the mocks at the end of the test method invocation. As a result, `#expect` works like an RSpec `allow` up until `#verify` is used to actually verify that calls were made.

In this PR, my goal is to add `#verify` calls to every mock. The only ones that don't have `#verify` are the really funky ones inside instance level method definitions, and for those which aren't the focus on what's being verified, I'm ok with them working like "allow".